src/utils/gitHubUtils.ts

```diff
@@ -105,3 +105,33 @@ export async function createFork(context: IActionContext, remoteRepo: ReposGetRe
         throw new Error(localize('forkFail', 'Could not automatically fork repository. Please fork [{0}]({1}) manually.', remoteRepo.name, remoteRepo.html_url));
     }
 }
+
+export async function createRepoFromTemplate(context: IActionContext, templateRepo: RepoData, newRepoName?: string, isPrivate?: boolean, description?: string): Promise<ReposCreateUsingTemplateResponse> {
```

Comment:
I think we can just include this in the `RepoCreateFromTemplateStep`.  Assuming we refactor it how I said in the comment below, it won't have to be called in two separate places.